sql/parser/eval.go

```diff
@@ -259,6 +263,156 @@ func (d DString) String() string {
 	return StrVal(d).String()
 }
 
+// maxTime and minTime are not the theoretical max/min for the time.Time
```

Comment:
It also matters that the overflow of `time.Time` occurs before the overflow of `time.Time.Unix`. This holds, but I don't think your code snippet is proof of that. Regardless, I agree that a definition of `IsMax` that looks like this should work:

```
func (d DDate) IsMax() bool {
  n := d.Add(1)
  return n.Before(d)
}
```

We could also pre-compute the max: http://play.golang.org/p/SwnnuNc_Jf

```
219248499-12-06 15:30:07.999999999 +0000 UTC
```

Adding 1 nanosecond to the above date results in the expected date string, but `Time.Before` says it is earlier because the `Time.sec` field wrapped around.